ui/widgets/nuxeo-date-picker.js

```diff
@@ -245,10 +245,10 @@ import { I18nBehavior } from '../nuxeo-i18n-behavior.js';
     }
 
     _getValidity() {
-      if (!this.required) {
-        return true;
+      if (!this.$.date.validate()) {
+        return false;
       }
-      return !!this.value;
+      return this.required ? !!this.value : true;
```

Comment:
Could be written in one line:
`return this.$.date.validate() && (this.required ? !!this.value : true);`